pkg/odo/cli/list/list.go

```diff
@@ -0,0 +1,284 @@
+%[1]s
+  `)
+
+// ListOptions ...
+type ListOptions struct {
+	// Context
+	*genericclioptions.Context
+
+	// Clients
+	clientset *clientset.Clientset
+
+	// Local variables
+	project         string
+	namespaceFilter string
+	devfilePath     string
+	localComponent  component.OdoComponent
+
+	// Flags
+	namespaceFlag string
+}
+
+// NewListOptions ...
+func NewListOptions() *ListOptions {
+	return &ListOptions{}
+}
+
+func (o *ListOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+// Complete ...
+func (lo *ListOptions) Complete(cmdline cmdline.Cmdline, args []string) (err error) {
+
+	// Check to see if KUBECONFIG exists, and if not, error the user that we would not be able to get cluster information
+	// Do this before anything else, or else we will just error out with the:
+	// invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
+	// instead
+	if !dfutil.CheckKubeConfigExist() {
+		return errors.New("KUBECONFIG not found. Unable to retrieve cluster information. Please set your Kubernetes configuration via KUBECONFIG env variable or ~/.kube/config")
+	}
+
+	// Create the local context and initial Kubernetes client configuration
+	lo.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline))
+	if err != nil {
+		return err
+	}
+
+	// Check for the Devfile and then retrieve all information regarding the local Devfile
+	lo.devfilePath = location.DevfileLocation("")
+	if util.CheckPathExists(lo.devfilePath) {
+
+		// Set the project / namespace based on the devfile context
+		lo.project = lo.Context.GetProject()
+
+		// Parse the devfile
+		devObj, parseErr := devfile.ParseAndValidateFromFile(lo.devfilePath)
+		if parseErr != nil {
+			return parseErr
+		}
+
+		// Create a local component from the parse devfile
+		localComponent := component.OdoComponent{
+			Name:      devObj.Data.GetMetadata().Name,
+			ManagedBy: "",
+			Modes:     map[string]bool{},
+			Type:      component.GetComponentTypeFromDevfileMetadata(devObj.Data.GetMetadata()),
+		}
+
+		lo.localComponent = localComponent
+
+	}
+
+	// If the context is "" (devfile.yaml not found..), we get the active one from KUBECONFIG.
+	if lo.project == "" {
+		lo.project = lo.clientset.KubernetesClient.GetCurrentNamespace()
+	}
+
+	// If the namespace flag has been passed, we will search there.
+	// if it hasn't, we will search from the default project / namespace.
+	if lo.namespaceFlag != "" {
+		lo.namespaceFilter = lo.namespaceFlag
+	} else {
+		lo.namespaceFilter = lo.project
+	}
+
+	return
+}
+
+// Validate ...
+func (lo *ListOptions) Validate() (err error) {
+	return nil
+}
+
+// Run has the logic to perform the required actions as part of command
+func (lo *ListOptions) Run(ctx context.Context) error {
+
+	// All variables pertaining to output / combination
+	devfileComponents := []component.OdoComponent{}
```

Comment:
This is not necessary (and make golangci-lint fail), because you are redefinig the variable below